go.mod

```diff
@@ -0,0 +1,5 @@
+module github.com/go-misc/async-retry
```

Comment:
[nits]

How about `github.com/Kyash/async-retry` as a module name?
I don't have a strong opinion, so I'll leave it to you to decide if you want to modify it or not.

Reply:
This is just a trash made when I developed at local environment. Thanks!

Reply:
Thanks for the quick fix!
Please also fix the sample code listed in the README

https://github.com/Kyash/async-retry/blob/first-impl/README.md?plain=1#L26